server/src/main/java/org/opensearch/gateway/remote/RemoteGlobalMetadataManager.java

```diff
@@ -217,4 +248,20 @@ private void setGlobalMetadataUploadTimeout(TimeValue newGlobalMetadataUploadTim
     public TimeValue getGlobalMetadataUploadTimeout() {
         return this.globalMetadataUploadTimeout;
     }
+
+    public RemoteClusterStateBlobStore<CoordinationMetadata, RemoteCoordinationMetadata> getCoordinationMetadataBlobStore() {
+        return coordinationMetadataBlobStore;
+    }
+
+    public RemoteClusterStateBlobStore<Settings, RemotePersistentSettingsMetadata> getPersistentSettingsBlobStore() {
+        return persistentSettingsBlobStore;
+    }
+
+    public RemoteClusterStateBlobStore<TemplatesMetadata, RemoteTemplatesMetadata> getTemplatesMetadataBlobStore() {
+        return templatesMetadataBlobStore;
+    }
+
+    public RemoteClusterStateBlobStore<Custom, RemoteCustomMetadata> getCustomMetadataBlobStore() {
+        return customMetadataBlobStore;
+    }
```

Comment:
why do you need get methods?

Reply:
Need to pass BlobStore for `getAsyncMetadataWriteAction` from RemoteClusterStateService.

Reply:
you can create a simple map registry based on entityType

Reply:
Created a registry like this and using that to get remote store.